inst/examples/02-components.Rmd

```diff
@@ -230,11 +230,21 @@ Theorems\index{theorem} and proofs are commonly used in articles and books in ma
 In **bookdown**, the types of theorem environments supported are in Table \@ref(tab:theorem-envs). To write a theorem, you can use the syntax below:
 
 ````markdown
-`r ''````{theorem}
-Here is my theorem.
-```
+::: theorem
+This is a `theorem` environment which can contain **any** _markdown_ syntax
+:::
 ````
 
+or the longer version
+
+````markdown
+::: {.theorem}
+This is a second theorem
+:::
+````
+
+This syntax is based on Pandoc's [fenced `Div` blocks](https://pandoc.org/MANUAL.html#divs-and-spans) and can already be used in any R Markdown document to write [Custom Block](https://bookdown.org/yihui/rmarkdown-cookbook/custom-blocks.html). **bookdown** is only offering special handling for Theorems and Proofs environments. One important advantage of this syntax is that it is Pandoc's Markdown and you can write any valid markdown inside the block.
```

Comment:
Punctuations need to be inside `[]` because links to be converted to footnotes in PDF, and we want footnote numbers to appear after (instead of before) punctuation marks.

Reply:
Oh sure. Thanks for the feedback!

Reply:
I remember Pandoc has an option to do that automatically, but I've never used it (perhaps we should).

Reply:
I don't think I know this option that would handle the `.`. Or are you meaning that we don't use Pandoc itself to transform links to footnotes ? I see the `links-as-notes` variables is not used in `index.Rmd`. 
How does the transformation works currently ?

Reply:
I just looked it up in the manual. I think I was referring to the `notes-after-punctuation` option: https://pandoc.org/MANUAL.html#other-relevant-metadata-fields but you are right that since we didn't use `links-as-notes`, this option probably won't work. Even if we use the `link-as-notes` option, I don't know if `notes-after-punctuation` will work, since it seems to be designed for citations.

Reply:
I did not knew that ones - so many extensions! I understand the same : It seems designed for citations anyway.